### ISSUE TITLE
Nightly builds with latest checkout of all commands instead of waiting for release for individual commands

### ISSUE THREAD

**Issue by Nikschavan** [wp-cli/wp-cli-bundle#354]
## Feature Request

 

- [x] Yes, I reviewed the [contribution guidelines](https://make.wordpress.org/cli/handbook/contributing/).

**Describe your use case and the problem you are facing**

Currently, Nightly builds are built when something is merged to the bundle repo. Since the development of all the actions happens in separate repos, they get added to nightly only after a release is created on the individual repos.

**Describe the solution you'd like**

Can the nightly builds be made to include master branches of all the individual commands?

One way of doing this I am imagining is - Create a separate branch on this repo from which nightlies can be built, This branch will need will need two changes - 1. drop prefer-stable from composer.json and 2. remove composer.lock to allow downloading latest versions.

One drawback I can think of is - Creating a separate branch is a maintenance task to keep this branch up to date with all the latest changes from main branch.

**Comment by schlessera** [comment]
I think the moving parts are just too many to get the tests to still pass if we're pulling in non-stable command repositories. Right now, the build of the nightly requires the smoke tests of the bundle to pass, and already with stable-only dependencies, this condition is often not met.

**Comment by schlessera** [comment]
What I want to get to is that command packages are more often released. I've already done a lot of work in streamlining all sorts of processes that go into a package release, and as you can see from the latest release, often single fixes even get their separate release.
However, this is often slowed down on new PHP major releases, as they contain countless breaking changes nowadays, which need fixes across multiple layers of the WP-CLI ecosystem.
